realtime/utils.js

```diff
@@ -400,7 +400,16 @@ function isIpWhitelisted(addr, whitelist) {
  */
 function attachAspectSubject(sample, useSampleStore, subjectModel,
   aspectModel) {
-  const nameParts = sample.name.split('|');
+  let nameParts;
+
+  // if name has not '|' then throw error
+  try {
+    nameParts = sample.name.split('|');
+  } catch (err) {
+    return Promise.resolve({ isFailed: true,
```

Comment:
Where does this Promise.resolve end up? ... logged?
If this is an unexpected and undesired state, shouldn't it be a Promise.reject and get handled and logged somewhere?